src/Marcher.hs

```diff
@@ -139,24 +139,37 @@ type Direction = Vec3
 rayRender :: ImageSettings -> Scene -> Ray -> Color
 rayRender sett s ray = clamp $ case rayMarch sett s ray of
     Nothing -> getBackgroundColor sett
-    Just pos -> let color = getColor $ snd $ s pos
-              in case rayMarch sett newscene (pos + 3*epsilon `scale` calcNormal sett s pos, normalize $ light - pos) of
-        Nothing -> color
-        Just newpos -> if equalWithinError epsilon 0 $ mag (light-newpos) then color else gray * color
-
-  where light = getSunPosition sett
-        newscene = mergeScenes s $ pointToScene light
-        epsilon = getTolerance sett
+    Just color -> color
 
 -- | Marches a ray through a scene until it hits an object. Returns Nothing if it goes outside the scene.
-rayMarch :: ImageSettings -> Scene -> Ray -> Maybe Position
+rayMarch :: ImageSettings -> Scene -> Ray -> Maybe Color
```

Comment:
Separating shading from pure marching was the entire point of having both `rayMarch` and `rayRender`

Reply:
Yeah, I just experimented with the shading and it was easier for me to put everything into one function.

Reply:
Anyway, I'll refactor it.